From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_linux@m4x.org>
Date: Sun, 10 Apr 2022 20:11:15 +0200
Subject: [PATCH] {MAYBE UPS} drivers/misc/habanalabs: cast to uintptr_t before
 u64
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Compiling on x86-32 with gcc fails:

    drivers/misc/habanalabs/common/memory.c: In function
    ‘alloc_device_memory’:
    drivers/misc/habanalabs/common/memory.c:153:49: error: cast from
    pointer to integer of different size [-Werror=pointer-to-int-cast]
      153 |                                                 (u64) gen_pool_dma_alloc_align(vm->dram_pg_pool,
          |                                                 ^
---
 drivers/misc/habanalabs/common/memory.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/misc/habanalabs/common/memory.c b/drivers/misc/habanalabs/common/memory.c
index e008d82e4ba3..5e07f73083de 100644
--- a/drivers/misc/habanalabs/common/memory.c
+++ b/drivers/misc/habanalabs/common/memory.c
@@ -150,7 +150,7 @@ static int alloc_device_memory(struct hl_ctx *ctx, struct hl_mem_in *args,
 		for (i = 0 ; i < num_pgs ; i++) {
 			if (is_power_of_2(page_size))
 				phys_pg_pack->pages[i] =
-						(u64) gen_pool_dma_alloc_align(vm->dram_pg_pool,
+						(u64)(uintptr_t) gen_pool_dma_alloc_align(vm->dram_pg_pool,
 										page_size, NULL,
 										page_size);
 			else
-- 
